### ISSUE TITLE
updates to gcds-input in regards to the addition of gcds-date-input

### ISSUE THREAD

**Issue by theonlyguills** [cds-snc/gcds-components#675]
Hi,

We've noticed the removal of the 'date' and 'time' type option for the gcds-input component following the addition of gcds-date-input.

Last I tried it seemed to still be supported at the code level although your documentation and examples removed it. Is it being phased out? We are using it extensively and would like to keep using it.

If it is being removed, what is the reason behind it? 

Thanks!

**Comment by daine** [comment]
@theonlyguills thanks for reaching out! 

What version of the components package were you using? As far as I'm aware, we have never supported `date` nor `time` as valid values for the `type` property on `<gcds-input>`. You mentioned documentation and examples however we're unable to find them in older versions, and this is visible in our git commit history.

We're aware that the regular HTML tag `<input>` has support for `date` and `time` but this is not the case for `<gcds-input>`. I hope this clarifies things a bit!

**Comment by theonlyguills** [comment]
Hi!

I'm curious. We're currently using version 0.22.x of gcds. We use it like

```
      <gcds-input
        id="date"
        for="date"
        type="date"
        :input-id="ids.date"
        :lang="$i18n.locale"
        :error-message="errorMessages.date"
        :value="models.date"
        @gcdsChange="emit('onDateUpdate', $event)"
      />

      <gcds-input
        id="time"
        for="time"
        type="time"
        :input-id="ids.time"
        :lang="$i18n.locale"
        :error-message="errorMessages.time"
        :value="models.time"
        @gcdsChange="emit('onTimeUpdate', $event)"
      />
```

It's providing us with a calendar picker that looks like 
![image](https://github.com/user-attachments/assets/e006b2fa-7ec0-4ec0-86b2-a48415017b12)

and a time picker that looks like
![image](https://github.com/user-attachments/assets/0655e5a0-1c07-48e9-a5a7-cb73e46a830e)

I have tried upgrading to the latest version and it still worked, even if it was removed from the doc.

Thoughts?

**Comment by daine** [comment]
@theonlyguills thank you, this information is very helpful! 

We have a select set of supported options for our components, in this case for the `<gcds-input>` component we only officially support the following types `"email" | "number" | "password" | "search" | "text" | "url"` ([see README](https://github.com/cds-snc/gcds-components/blob/main/packages/web/src/components/gcds-input/readme.md)). That said, I looked into it further and there are currently no additional hard checks for it. I tested it out and you are correct, it passes through to the underlying `<input>` HTML tag which then renders it according to the browser/system.

As of now, this is an unsupported use case, however, I'm taking note of it and will bring it back to our team. I will close this ticket for now, and if there will be any changes or decisions on this we'll make sure to give you a heads up either through our mailing list or here.
